# Classification of Inappropriate/Offensive Text

## Import the essentials

In [20]:
import pandas as pd
import numpy as np

# Load the dataset
data = pd.read_csv('dataset.csv')
data.head()

text  label
0  Gerçekten sizin hikayelerinizi izleyerek mi ye...      0
1            @USER Çoook çok bi baklava bi sen zaten      0
2  1) Sn. DÜKEL; Atatürk'ün, Karma E. M. ile başl...      0
3  Konfederasyonumuzun Aile ve Sosyal Politikalar...      0
4  Hakemler tarih yazıyorlar / 9 kişiye karşı 3-2...      1

## Data Preprocessing Part

In [22]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 53005 entries, 0 to 53004
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   text    53005 non-null  object
 1   label   53005 non-null  int64 
dtypes: int64(1), object(1)
memory usage: 828.3+ KB


In [21]:
# Function to remove unnecessary user tags from the entries
def clean_text(text):
    text = text.replace("@USER ", "")
    text = text.replace("@USER", "")
    return text

# Apply the cleaning function to the 'text' column
data['text'] = data['text'].apply(clean_text)

# Remove emojis from the samples
data.astype(str).apply(lambda x: x.str.encode('ascii', 'ignore').str.decode('ascii'))

# Save the cleaned dataframe
data.to_csv('cleaned_dataset.csv', index=False)

In [17]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 53005 entries, 0 to 53004
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   text    53005 non-null  object
 1   label   53005 non-null  int64 
dtypes: int64(1), object(1)
memory usage: 828.3+ KB


## Exploratory Data Analysis Results
The dataset includes 53005 rows and is about to offensive text. We can clearly see that our data has 2 columns which are not null: *text* and *label*. So we don't need to clear or fill the null values.

count      mean       std  min  25%  50%  75%  max
label  53005.0  0.463692  0.498685  0.0  0.0  0.0  1.0  1.0